stdlib/asm/crypto/stark/verifier.masm

```diff
@@ -0,0 +1,1439 @@
+use.std::crypto::fri::frie2f4
+use.std::crypto::fri::ext2fri
+
+use.std::crypto::stark::deep_queries
+use.std::crypto::stark::random_coin
+
+
+
+# <= [log2(domain_size), ..]
+# => [domain_gen, ..]
+export.compute_generator
+    push.32
+    swap
+    sub
+    pow2
+    push.7277203076849721926
+    swap
+    exp.u64
+    # => [domain_gen, ..]
+end
+
+# <= [lde_size, 1, ..]
+# => [remainder_size, lde_size, num_fri_layers,..]
+export.get_num_fri_layers
+    dup
+    push.4
+    u32checked_divmod
+
+    dup.1
+    push.0
+    neq
+
+    while.true
+        drop
+        movup.2
+        add.1
+        movdn.2
+
+        push.4
+        u32checked_divmod
+
+        dup.1
+        push.64
+        u32unchecked_gt
+    end
+    drop
+end
+
+#<= [a, ..]
+#=> [log2(a), ..]
+proc.log2
```

Comment:
I think this is needed only because I use `trace_length` instead of `log(trace_length)` as input to the STARK verifier procedure. Once that's changed, the above procedure will not be needed as far as I can see.